### PR TITLE
made TypeNotFoundException accept LocatableString instead of str

### PR DESCRIPTION
**Pull request by sanderr**

### CODE REVIEW

File: src/inmanta/ast/__init__.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def resolve(self) -> Location:
 
 
 class TypeReferenceAnchor(Anchor):
-    def __init__(self, range: Range, namespace: "Namespace", type: str) -> None:
+    def __init__(self, range: Range, namespace: "Namespace", type: LocatableString) -> None:
         Anchor.__init__(self, range=range)
```
Comment (wouterdb) [inline src/inmanta/ast/__init__.py]: do we still need the range if the type is locatable?

File: src/inmanta/ast/statements/define.py [hunk 2]
```diff
@@ -102,10 +103,10 @@ def __init__(
         else:
             self.comment = None
 
-        self.parents = [str(x) for x in parents]
+        self.parents = parents
 
         if len(self.parents) == 0 and not (self.name == "Entity" and self.namespace.name == "std"):
-            self.parents.append("std::Entity")
+            self.parents.append(LocatableString("std::Entity", Range("__internal__", -1, -1, -1, -1), -1, namespace))
```
Comment (wouterdb) [inline src/inmanta/ast/statements/define.py]: perhaps we should move the Range to a constant?

File: src/inmanta/ast/__init__.py (hunk 1 above)
Comment (sanderr) [reply src/inmanta/ast/__init__.py]: Valid point, I'll look into it

**Comment by sanderr** [comment]
TODO: test case for #1221

File: src/inmanta/ast/statements/define.py (hunk 2 above)
Comment (sanderr) [reply src/inmanta/ast/statements/define.py]: Do you mean a constant that can be imported from `inmanta.ast`?

File: src/inmanta/plugins.py [hunk 3]
```diff
@@ -264,6 +264,12 @@ def to_type(self, arg_type, resolver):
                 "bad annotation in plugin %s::%s, expected str but got %s (%s)"
                 % (self.ns, self.__class__.__function_name__, type(arg_type), arg_type)
             )
+        filename: Optional[str] = inspect.getsourcefile(self.__class__.__function__)
+        location: Range
+        assert filename is not None
+        line: int = inspect.getsourcelines(self.__class__.__function__)[1] + 1
+        location = Range(filename, line, 1, line, -1)
```
Comment (wouterdb) [inline src/inmanta/plugins.py]: not sure of the -1 is right, perhaps take start position + 1

**Pull request merged**